docs/linker-attributes.md

```diff
@@ -0,0 +1,78 @@
+# Linker attributes
+
+As the linker scans the referenced code, it warns about non-understood reflection patterns or other code patterns whose behavior may not be preserved by linking. The following attributes influence the pattern-detection and warning behavior.
+
+## DynamicallyAccessedMembers
+
+Gives extra information about dynamic access patterns which would not otherwise be understood. This can tell the linker to keep additional members that are reflected over, making simple reflection patterns linker-friendly.
+
+TODO: provide examples
+
+The attribute specifies which members of a type are dynamically accessed (e.g. by reflection or interop code). It can be applied to IL locations (method parameters, method return parameters, fields, and properties) that hold a representation of a type, and it indicates via DynamicallyAccessedMemberTypes which members of that type are used dynamically. The linker uses information from these attributes to keep additional members of types, which can make some reflection access patterns linker-safe, eliminating the corresponding warnings.
+
+See [Cross-method annotations](design/reflection-flow.md#Cross-method-annotations) for more information.
+
+## RequiresUnreferencedCode
+
+Specifies that a method requires code which is not statically referenced, preventing analysis of the method, and warning for calls to it.
+
+TODO: provide examples
+
+When annotating linker-unfriendly code, this attribute can be used to prevent multiple warnings originating from reflection patterns in the same method body, instead turning them into a single warning with a message that explains why the method is conceptually linker-unsafe. Calls to methods with this attribute will result in a single warning for the callsite.
+
+## DynamicDependency
+
+Indicates that a member has a dependency on other members. This results in the referenced members being kept whenever the member with the attribute is kept. It does not otherwise influence the analysis behavior.
+
+```csharp
+using System;
+using System.Reflection;
+using System.Diagnostics.CodeAnalysis;
+
+class Program
+{
+    [DynamicDependency("Helper")]
```

Comment:
Id' be better to use some tricky code which linker cannot automatically detect as the example

Reply:
Linker will not figure out the sample below - we don't handle `GetExecutingAssembly` - but I guess it "looks obvious" - so maybe something less obvious...